packages/crypto/src/getAddresses.ts

```diff
@@ -0,0 +1,85 @@
+import { BtcAddress } from '@btckit/types';
+import { BitcoinAccount, ecdsaPublicKeyToSchnorr } from '@leather-wallet/bitcoin/src/bitcoin.utils';
+import { NetworkConfiguration } from '@leather-wallet/constants';
+import { bytesToHex } from '@stacks/common';
+import { TransactionPayload } from '@stacks/connect';
+
+import { useCurrentAccountNativeSegwitIndexZeroSigner } from './utils/bitcoin/native-segwit-account.hooks';
+import { useCurrentAccountTaprootIndexZeroSigner } from './utils/bitcoin/taproot-account.hooks';
+import { useCurrentStacksAccount } from './utils/stacks/stacks-account';
+import { StacksAccount } from './utils/stacks/stacks-account.models';
+import { KeyConfig } from './utils/useWalletType';
+
+export function useGetAddresses({
+  signatureIndex,
+  currentAccountIndex,
+  currentNetwork,
+  hasLedgerKeys,
+  hasSwitchedAccounts,
+  segwitAccount,
+  stacksAccounts,
+  taprootAccount,
+  transactionPayload,
+  wallet,
+}: {
+  signatureIndex: number | undefined;
+  currentAccountIndex: number;
+  currentNetwork: NetworkConfiguration;
+  hasLedgerKeys: boolean;
+  hasSwitchedAccounts: boolean;
+  segwitAccount: BitcoinAccount | undefined;
+  stacksAccounts: StacksAccount[];
+  taprootAccount: BitcoinAccount | undefined;
+  transactionPayload: TransactionPayload | null;
+  wallet: KeyConfig | undefined;
```

Comment:
Confusing naming as it's specific to stacks. Only `@leather/stacks` should know of these details